docs/discovery/provisioning.md

```diff
@@ -0,0 +1,456 @@
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 client secret").
+          SetDescription("OAuth2 client secret.").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 provider").
+          SetDescription("URL provider of access tokens. Example: http://mysite.com/oauth/token").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 mode").
+          SetDescription("Whether to send client credentials in body or via header of the request.").
+          SetRequired().
+          IsString().
+          SetEnumValues([]string{"header", "body"})).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 scope").
+          SetDescription("Scope supported for the client credentials OAuth2 authorization. Example: READ").
+          IsString())).
+    AddItem(apic.NewSubscriptionSchemaPropertyBuilder().
+      SetName("Credentials").
+      IsObject().
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("Username").
+          SetDescription("Your username credential.").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("Password ").
+          SetDescription("Your password credential.").
+          SetRequired().
+          IsString()))
+```
+
+## Handle Provisioning Requests
+
+The Agents SDK provides a registration method for the agent to implement an interface that is called when certain provisioning events happen in Amplify Central. Once registered the Agents SDK will watch for these events and trigger the appropriate method for the agent o handle that provisioning request.
+
+### Creating and Registering Provisioner
+
+This is the provisioning interface that the agent is expected to implement.
+
+```go
+// Provisioning - interface to be implemented by agents for access provisioning
+type Provisioning interface {
+  ApplicationRequestProvision(ApplicationRequest) RequestStatus
+  ApplicationRequestDeprovision(ApplicationRequest) RequestStatus
+  AccessRequestProvision(AccessRequest) RequestStatus
+  AccessRequestDeprovision(AccessRequest) RequestStatus
+  CredentialProvision(CredentialRequest) (RequestStatus, Credential)
+  CredentialDeprovision(CredentialRequest) RequestStatus
+}
+```
+
+Once implemented the agent may register the structure that implements it with the following call.
+
+```go
+import "github.com/Axway/agent-sdk/pkg/agent"
+
+agent.RegisterProvisioner(provisioningImplementation)
+```
+
+#### RequestStatus return structure
+
+All Provisioning request methods will expect a RequestStatus object to be returned after handing the request.  To create this object the Agent SDK provides a builder with the following methods.
```

Comment:
comma here...
To create this object, the Agent SDK provides a builder with the following methods.